lib/Backend/GlobOpt.cpp

```diff
@@ -7184,13 +7198,36 @@ GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
     if (prevSymStore && prevSymStore->IsStackSym() &&
         prevSymStore->AsStackSym()->HasByteCodeRegSlot())
     {
-        Assert(this->blockData.changedSyms != nullptr);
-        this->blockData.changedSyms->Set(prevSymStore->m_id);
+        this->SetChangedSym(prevSymStore->m_id);
     }
     valueInfo->SetSymStore(sym);
 }
 
 void
+GlobOpt::SetChangedSym(SymID symId)
+{
+    GlobOptBlockData * globOptData = nullptr;
+    if (this->currentBlock->globOptData.changedSyms)
+    {
+        globOptData = &this->currentBlock->globOptData;
+    }
+    else
```

Comment:
This is a bit ugly...  Is the only call where this is needed from MergeValues()?  We could just directly set the valueInfo in this case, because your cross-block code already handle the merge here...